src/providers/authProvider/permissions.ts

```diff
@@ -0,0 +1,111 @@
+type UserRoles = UserRole | 'user'
+
+const permissions: Record<UserRoles, ResourcePermissions> = {
+  user: {
+    projects: { read: true, write: false, delete: false },
+    batches: { read: true, write: false, delete: false },
+    items: { read: true, write: false, delete: false },
+    'reference-data': { read: true, write: false, delete: false }
+  },
+  'rco-user': {
+    projects: { read: true, write: true, delete: false },
+    batches: { read: true, write: true, delete: false },
+    items: { read: true, write: true, delete: false },
+    users: { read: true, write: true, delete: false },
+    platforms: { read: true, write: true, delete: false },
+    'reference-data': { read: true, write: true, delete: false }
+  },
+  'rco-power-user': {
+    '*': { all: '*' }
+  }
+}
+
+export default permissions
+
```

Comment:
@salony75way - for the following method, I wish to understand the two parameters.

Is the first parameter the set of permissions being requested, and the second parameter the responses to those requests?

Reply:
Hi @IanMayo,
`generatePermission` method is a helper for `mergedPermission` to merge permission if user has multiple permissions.

`generatePermission` method takes two arguments, first is permission for resource  and second is merged permissions object.